pkg/gohai/utils/value_helpers.go

```diff
@@ -0,0 +1,57 @@
+// This file is licensed under the MIT License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2014-present Datadog, Inc.
+
+package utils
+
+import "strconv"
+
+// ValueErrorGetter returns a function to get the error from the Value
+func ValueErrorGetter[T any](value *Value[T]) func() error {
```

Comment:
Why not returning the error directly?

Reply:
<-- agree with a suggestion -->
**👍 agreed:** ‏I guess replacing it with an `Error()` function would simplify the code (see https://github.com/DataDog/datadog-agent/pull/18073)